examples/with_winit/src/lib.rs

```diff
@@ -451,8 +461,9 @@ fn run(
                             .surface
                             .get_current_texture()
                             .expect("failed to get surface texture");
-                        #[cfg(not(target_arch = "wasm32"))]
-                        {
+                        // Note: we don't run the async/"robust" pipeline, as
+                        // it requires more async wiring for the readback.
```

Comment:
Could we get more detail here? What is meant by "async wiring" - Is this a limitation of !Send futures, single-threaded wasm, etc. - What *exactly* is the issue on WASM that we aren't doing async pipelines there?

Reply:
The thread is [#gpu > async on wasm](https://xi.zulipchat.com/#narrow/stream/197075-gpu/topic/async.20on.20wasm). Broadly, the issue is: if we spawn a future, that needs to keep hold of the `scene`. But that's incompatible with lifetimes of keeping the scene around (to reuse allocations, etc.)

In theory, we should do this much more carefully, but we haven't yet. That might also be useful in contexts outside of browsers too, allowing multi-threaded scene rendering